freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -20,13 +20,112 @@
+from http import HTTPStatus
+from requests.adapters import HTTPAdapter
+
+from freshmaker import conf, db, log
+from freshmaker.errata import Errata
 from freshmaker.events import FlatpakModuleAdvisoryReadyEvent
+from freshmaker.handlers import ContainerBuildHandler, fail_event_on_handler_exception
+from freshmaker.kojiservice import koji_service
+from freshmaker.lightblue import LightBlue
+from freshmaker.models import Event
+from freshmaker.pyxis import Pyxis
+from freshmaker.types import EventState
 
 
 class RebuildFlatpakApplicationOnModuleReady(ContainerBuildHandler):
     name = 'RebuildFlatpakApplicationOnModuleReady'
 
     def can_handle(self, event):
         return isinstance(event, FlatpakModuleAdvisoryReadyEvent)
+
+    @fail_event_on_handler_exception
+    def handle(self, event):
+
+        if event.dry_run:
+            self.force_dry_run()
+
+        self.event = event
+
+        db_event = Event.get_or_create_from_event(db.session, event)
+        self.set_context(db_event)
+
+        auto_build_images_list = self._get_auto_rebuild_image_list(event)
+        if not auto_build_images_list:
+            msg = ("There is no auto rebuild image can be rebuilt. "
+                   f"message_id: {event.msg_id}")
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        rebuild_images = self._filter_images_with_higher_rpm_nvr(event, auto_build_images_list)
+        if not rebuild_images:
+            msg = ("There is no image with higher rpm nvr can be rebuilt. "
+                   f"message_id: {event.msg_id}")
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        return []
+
+    def _get_auto_rebuild_image_list(self, event):
+        """
+        Get rebuild image list which can be auto rebuilt.
+
+        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
+            is currently handling.
+        :rtype: list
+        :return: List of images which can be auto rebuilt.
+        """
+        errata = Errata()
+        module_nvrs = errata.get_cve_affected_build_nvrs(event.advisory.errata_id, True)
+        rebuild_images_list = list()
+        self._pyxis = Pyxis(conf.pyxis_server_url)
+
+        with koji_service(conf.koji_profile, log, login=False, dry_run=self.dry_run) as session:
+            for module_nvr in module_nvrs:
+                mmd = session.get_modulemd(module_nvr)
+                content_index_url = '{}/released/contents/modules/{}:{}.json'.format(conf.flatpak_server_url,
+                                                                                     mmd.get_module_name(),
+                                                                                     mmd.get_stream_name())
+                session = requests.Session()
+                adapter = HTTPAdapter(max_retries=3)
```

Comment:
Oh, this does not handle the actual HTTP request timeouts - from [docs](https://2.python-requests.org/en/master/api/#requests.adapters.HTTPAdapter):

> max_retries – The maximum number of retries each connection should attempt. Note, this applies only to failed DNS lookups, socket connections and connection timeouts, never to requests where data has made it to the server. By default, Requests does not retry failed connections. If you need granular control over the conditions under which we retry a request, import urllib3’s Retry class and pass that instead.

See a good example of a well-tested retry implementation from cachito: https://github.com/release-engineering/cachito/blob/db1acca3f5c56add57d7f82b55c44122e93ea394/cachito/workers/requests.py#L48

Reply:
Ack. Fixing